File: src/dotnet/Common/Models/Chat/MessageHistoryItem.cs
using Newtonsoft.Json;

namespace FoundationaLLM.Common.Models.Chat
{
    /// <summary>
    /// Represents an historic message sender and text item.
    /// </summary>
    public class MessageHistoryItem
    {
        /// <summary>
        /// The sender of the message (e.g. "Agent", "User").
        /// </summary>
        [JsonProperty("sender")]
        public string Sender { get; set; }
        /// <summary>
        /// The message text.
        /// </summary>
        [JsonProperty("text")]
        public string Text { get; set; }

        /// <summary>
        /// Message history item
        /// </summary>
        /// <param name="sender">The sender of the message (e.g., "Agent", "User")</param>
        /// <param name="text">The message text.</param>
        public MessageHistoryItem(string sender, string text)
        {
            Sender = sender;
            Text = text;
        }
    }
}

File: src/python/PythonSDK/foundationallm/models/orchestration/message_history_item.py
from pydantic import BaseModel

class MessageHistoryItem(BaseModel):  
    """
        Represents an historic message sender and text item.
        
        Parameters
        ----------
        sender : str
            The sender of the message (e.g., "Agent", "User")
        text : str
            The message text.
    """    
    sender: str
    text: str

    